### PR TITLE
Small improvements & fixes to SWE-Bench

### PR DESCRIPTION
**Pull request by li-boxuan**
I was able to run a few benchmark instances from SWE-Bench by myself following the documentation - it was great! In general the experience was smooth, thanks to @xingyaoww, @libowen2121 and the team! I made a few small enhancements and fixes to further improve the developer experience.

1. Always use `poetry run python` (using python from poetry's virtual environment) over `python` or `python3` in scripts to make sure the behavior is consistent.
2. Make AGENT configurable. One can use an argument to control which agent they would like to benchmark. To facilitate this, I removed hardcoded CodeActAgent from `run_infer.sh`, and also added `VERSION` attribute to all agents, as the benchmark needs to record the agent version.
3. Make EVAL_LIMIT configurable. One can use an argument to control how many instances they'd like to benchmark. Useful for debugging & development purposes.
4. Fix 'eval_output_dir' not defined error in `run_infer.py`.
5. Other enhancements to the README file and logs.

I also notice that a lot of code from `run_infer.py` could be shared by other benchmarks, but since we only have one benchmark now, I think we could avoid over-engineering. A refactor and code dedup would be useful in the future once we have more benchmarks, though.

### CODE REVIEW

File: evaluation/swe_bench/README.md [hunk 1]
```diff
@@ -68,20 +68,41 @@ temperature = 0.0
...
-If you get to the interactive shell successfully, it means success!
+If you get to the interactive shell successfully, it means your environment works!
+If you see an error, please make sure your `config.toml` contains all
+`SWEBench eval specific` settings as shown in the previous section.
 
 ## Run Inference on SWE-Bench Instances
 
 ```bash
-./evaluation/swe_bench/scripts/run_infer.sh eval_gpt4_1106_preview
+./evaluation/swe_bench/scripts/run_infer.sh [model_config] [agent] [eval_limit]
```
Comment (xingyaoww) [inline evaluation/swe_bench/README.md:85]: Maybe we could provide an example command (commented out under this line)? like

`./evaluation/swe_bench/scripts/run_infer.sh eval_gpt4_1106_preview CodeActAgent 300` as an example?

**Review by xingyaoww — APPROVED** [review]
Looks awesome to me!! Thanks for polishing it so well! :)

File: evaluation/swe_bench/README.md (hunk 1 above)
Comment (li-boxuan) [reply evaluation/swe_bench/README.md:85]: See L104 if you missed it :)

Comment (xingyaoww) [reply evaluation/swe_bench/README.md:85]: Yep! Just next line maybe :D -- but it shouldn't matter too much

Comment (li-boxuan) [inline evaluation/swe_bench/README.md:85]:
```suggestion
./evaluation/swe_bench/scripts/run_infer.sh [model_config] [agent] [eval_limit]
# e.g., ./evaluation/swe_bench/scripts/run_infer.sh eval_gpt4_1106_preview CodeActAgent 300
```

Comment (li-boxuan) [reply evaluation/swe_bench/README.md:85]: make sense, added.

Comment (xingyaoww) [reply evaluation/swe_bench/README.md:85]: LGTM! Let's merge!

**Comment by libowen2121** [comment]
Hey @li-boxuan! Thanks for the excellent work!

**Pull request merged**